src/Compiler/Checking/import.fs

```diff
@@ -166,21 +166,115 @@ let CanImportILTypeRef (env: ImportMap) m (tref: ILTypeRef) =
 let ImportTyconRefApp (env: ImportMap) tcref tyargs nullness = 
     env.g.improveType tcref tyargs nullness
 
-let ImportNullness (g: TcGlobals) =
-    ignore g
-    // if g.langFeatureNullness && g.assumeNullOnImport then
-    //     KnownWithNull
-    // else
-    // TODO NULLNESS
-    KnownAmbivalentToNull
-
-let ImportNullnessForTyconRef (g: TcGlobals) (m: range) (tcref: TyconRef) =
-    ignore (g, tcref, m)
-    // if g.langFeatureNullness && g.assumeNullOnImport && TyconRefNullIsExtraValue g m tcref then
-    //     KnownWithNull
-    // else
-    // TODO NULLNESS
-    KnownAmbivalentToNull
+
+module Nullness =
+
+    open FSharp.Compiler.AbstractIL.Diagnostics
+
+    let arrayWithByte0 = [|0uy|]
+    let arrayWithByte1 = [|1uy|]
+    let arrayWithByte2 = [|2uy|]
+
+    let knownAmbivalent = NullnessInfo.AmbivalentToNull |> Nullness.Known
+    let knownWithoutNull = NullnessInfo.WithoutNull |> Nullness.Known
+    let knownNullable = NullnessInfo.WithNull |> Nullness.Known
+
+    let mapping byteValue = 
+        match byteValue with
+        | 0uy -> knownAmbivalent
+        | 1uy -> knownWithoutNull
+        | 2uy -> knownNullable
+        | _ ->
+            dprintfn "%i was passed to Nullness mapping, this is not a valid value" byteValue
+            knownAmbivalent         
+
+    let tryParseAttributeDataToNullableByteFlags (g:TcGlobals) attrData = 
+        match attrData with
+        | None -> ValueNone
+        | Some ([ILAttribElem.Byte 0uy],_) -> ValueSome arrayWithByte0
+        | Some ([ILAttribElem.Byte 1uy],_) -> ValueSome arrayWithByte1
+        | Some ([ILAttribElem.Byte 2uy],_) -> ValueSome arrayWithByte2
+        | Some ([ILAttribElem.Array(byteType,listOfBytes)],_) when byteType = g.ilg.typ_Byte -> 
+            listOfBytes
+            |> Array.ofList
```

Comment:
It's just a nitpick, but something to keep in mind - it will be allocating a bunch when checking for attributes.